File: first/php/library/holiday.php
<?php
class Holiday extends Extras {
    
    private float $holiday = 2.0;
    private int $factor;

    public function __construct(int $extraHours) {
        parent::__construct($extraHours);
        $this->factor = $extraHours;
    }

    public function extraPay(int $valueHour): float {
        return ($valueHour * $this->holiday) * $this->factor;
    }
}
?>

File: first/java/Holiday.java

public class Holiday extends Extras {

    private final float holiday = 2.0f;
    private final int factor;

    public Holiday(int extraHours) {
        super(extraHours);
        this.factor = extraHours;
    }

    @Override
    public float extraPay(int valueHour) {
        return (valueHour * holiday) * factor;
    }
}
